addons/mail/models/discuss/discuss_channel.py

```diff
@@ -646,6 +646,20 @@ def _notify_thread(self, message, msg_vals=False, **kwargs):
             ),
             (self, "discuss.channel/new_message", payload),
         ]
+        # notify mentioned internal users with access which are not member
+        if "partner_ids" in msg_vals:
+            member_partner_ids = self.env['discuss.channel.member'].with_context(active_test=False).search([
+                ('partner_id', '!=', self.env.user.partner_id.id),
+                ('channel_id', '=', self.id)
+            ]).mapped('partner_id.id')
+            for p in self.env["res.partner"].browse([pid for pid in msg_vals["partner_ids"] if pid not in member_partner_ids]):
+                bus_notifications.append((p, 'discuss.channel/mentioned', {
+                                'channel': {
+                                    **self._channel_basic_info(),
+                                    "is_pinned": True,
+                                },
+                                'mentioned_by_user_id': self.env.user.id,
+                }))
```

Comment:
Please don't add a new custom notification type. We had a lot of them and we standardized to `mail.record/insert`, so that the processing is consistent and follows the architecture around the store in Discuss.

If you want to display a notification, better make it as a side-effect of inserting in models. That way server-client code is all about exchanging model data, behaviour are consequence of model (changes).

FYI other custom notifications like `discuss.channel/joined` are technical debts: we haven't taken the time to convert them, they are not example to follow.